nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/impl/MSSQLDatabaseAdapter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.standard.db.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.processors.standard.db.DatabaseAdapter;
+
+/**
+ * A database adapter that generates MS SQL Compatible SQL.
+ */
+public class MSSQLDatabaseAdapter implements DatabaseAdapter {
+    @Override
+    public String getName() {
+        return "MS SQL";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Generates MS SQL Compatible SQL, for version 2012 or greater";
+    }
+
+    @Override
+    public String getSelectStatement(String tableName, String columnNames, String whereClause, String orderByClause, Long limit, Long offset) {
+        if (StringUtils.isEmpty(tableName)) {
+            throw new IllegalArgumentException("Table name cannot be null or empty");
+        }
+        final StringBuilder query = new StringBuilder("SELECT ");
+
+        //If this is a limit query and not a paging query then use TOP in MS SQL
+        if (limit != null && offset == null){
+            query.append("TOP ");
```

Comment:
@SQLGuyChuck `TOP` in this case is supposed to emulate a `FETCH` with `OFFSET` 0. In order to maintain that functionality I think WITH TIES would probably have unintended consequences, including data duplication in the next page of data (the tied record(s) would be in page 1 and it would also be the first record(s) in page 2).

Thoughts?

Also, if we wanted to add TOP WITH TIES style functionality in general that would be a separate ticket, as we'd need to touch every DatabaseAdapter (since they all do top/limit differently).